eip/EIPS/eip-aa-rules.md

```diff
@@ -0,0 +1,351 @@
+which means that unrelated `UserOperations` or transactions may invalidate each other.
+
+If not addressed, this would make the job of maintaining a mempool of valid `UserOperations` and producing valid
+bundles computationally infeasible and susceptible to DoS attacks.
+
+This document describes set of validation rules that, if applied by a block builder prior including a `UserOperation`,
+can prevent such attacks.
+
+## Specification
+
+### Definition of the `mass invalidation attack`
+
+A possible set of actions is considered to be a `mass invalidation attack` on the network if a large number of
+`UserOperations` that did pass the initial validation and were accepted by nodes and propagated further into the
+mempool to all bundlers in the network, becomes invalid and not eligible for inclusion in a block.
+
+There are 3 ways to perform such an attack:
+
+1. Create a code that passes the initial validation, but later fails the re-validation
+   that is performed during the bundle creation.
+2. Submit `UserOperations` that are valid in isolation during validation, but when bundled together become invalid.
+3. Submit valid `UserOperations` but "front-run" them by executing a state change on the
+   network that causes them to become invalid. The "front-run" in question must be economically viable.
+
+To prevent such attacks, we attempt to "sandbox" the validation code.
+We isolate the validation code from other `UserOperations`, from external changes to the storage, and
+from information about environment such as a current block timestamp.
+
+### What is not considered a `mass invalidation attacl`
+
+A `UserOperation` that fails the initial validation by a receiving node without entering its mempool is not
+considered an attack. The node is expected to apply web2 security measures and throttle requests based on API key,
+source IP address, etc.
+RPC nodes already do that to prevent being spammed with invalid transactions which also have a validation cost.
+
+### Constants:
+
+
+| Title                                | Value                       | Comment                                                                         |
+|--------------------------------------|-----------------------------|---------------------------------------------------------------------------------|
+| `MIN_UNSTAKE_DELAY`                  | 84600                       | 1 day                                                                           |
+| `MIN_STAKE_VALUE`                    | Adjustable per chain value  | Equivalent to ~$1000 in native tokens                                           |
+| `SAME_SENDER_MEMPOOL_COUNT`          | 4                           |                                                                                 |
+| `SAME_UNSTAKED_ENTITY_MEMPOOL_COUNT` | 10                          |                                                                                 |
```

Comment:
What is the rational for `10` here?

That value seems low enough that for any highly used paymaster or factory will need to stake. Is that the intention?

Reply:
it is small enough to prevent "mass-invalidation" attacks, but the rules below cause the actual value to rise rapidly (as long as UserOps gets included).